File: 0053-maximum-subarray/0053-maximum-subarray.go
func maxSubArray(nums []int) int {
	sum := 0
    maxi := math.MinInt64
	for _, val := range nums {
		sum += val

        if sum > maxi {
            maxi = sum
        }
        
		if sum < 0 {
			sum = 0
		}
        
	}
	return maxi
}

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
    let m = Number.MIN_SAFE_INTEGER, l = nums.length, sm = m;
    while (l--) {
        sm = Math.max(nums[l], sm + nums[l])
        m = Math.max(sm, m)
    }
    return m
};
